controllers/kamajicontrolplane_controller_tcp.go

```diff
@@ -179,6 +179,11 @@ func (r *KamajiControlPlaneReconciler) createOrUpdateTenantControlPlane(ctx cont
 			} else {
 				tcp.Spec.ControlPlane.Ingress = nil
 			}
+			// LoadBalancer
+			if kcp.Spec.Network.LoadBalancerConfig != nil {
+				tcp.Spec.NetworkProfile.LoadBalancerClass = kcp.Spec.Network.LoadBalancerConfig.LoadBalancerClass
```

Comment:
Maybe we should check if `kcp.Spec.Network.LoadBalancerConfig.LoadBalancerClass` is not nil and assign its value.

```suggestion
				if lbClass := kcp.Spec.Network.LoadBalancerConfig.LoadBalancerClass; lbClass != nil {
					tcp.Spec.NetworkProfile.LoadBalancerClass = 
ptr.To(*lbClass)
				}
```

Of course, we need to import `k8s.io/utils/ptr`

Reply:
Yes! good point. Addressed.